### ISSUE TITLE
useless error message when invalid settings in `ruff.toml`

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff-vscode#642]
```toml
[lint]
ignore = ["asdfasdf"]
```

this causes the following error message to appear:

> Error while resolving settings from workspace c:\Users\user\project. Please refer to the logs for more details.

however there's nothing  useful in the logs:

![image](https://github.com/user-attachments/assets/344511dc-0b26-43ee-b2ae-a090c681dbd3)

running `ruff check` in the cli shows the actual error:

```
> ruff check
ruff failed
  Cause: Failed to parse C:\Users\user\project\ruff.toml
  Cause: TOML parse error at line 6, column 1
  |
6 | [lint]
  | ^^^^^^
Unknown rule selector: `asdfasdf`
```

ideally this error would be included in the log

**Comment by KotlinIsland** [comment]
ideally this error would be included in the notification

**Comment by dhruvmanila** [comment]
Thanks for the issue. This is a deliberate choice from my end because of the way configuration works. The indexer runs in parallel and builds up the settings index collecting all the errors found during this process. Even if there are multiple files with an error, we'd only display a single popup notification and provide all the details in the output window. If we were to show a notification popup for every error, then it might spam the window with multiple of them.

One option would be to create a list of error messages and show the entire message in a single notification window but it might still create a rather large popup. You can already see that the message for a single error is 6 lines long. This was also the reason that we're providing the "Show Logs" button on the notification popup so that it's just 1-click away.

That said, I'm curious to know other's thoughts on this. I'm more than happy to reconsider if the latter is more useful from a user perspective. I chose this for a better UX :)

For reference, this is the method that I'm talking about:
https://github.com/astral-sh/ruff/blob/924741cb11a68ed037899f9db1bea6969c48385e/crates/ruff_server/src/session/index/ruff_settings.rs#L103-L117

**Comment by KotlinIsland** [comment]
what about: if there is a single error, or the combination of all error messages is less than a couple lines, then show it in the notification, if there is extra, say the message is large and have the "show log" button?

**Comment by MichaReiser** [comment]
We should definitely fix that the message isn't included in the logs by default

**Comment by dhruvmanila** [comment]
@MichaReiser By message, do you mean this one: "Failed to parse C:\Users\user\project\ruff.toml" ?

**Comment by MichaReiser** [comment]
Yes, we should at least show a message similar to the `ruff check` output (in the output panel)

**Comment by dhruvmanila** [comment]
Oh I see the issue. You'll need to enable server messages by setting `ruff.trace.server` to `messages` for them to show up in the output channel. I want to address this in the coming weeks.

**Comment by cschmidt0121** [comment]
Had the same issue here due to a missed typo in pyproject.toml. Definitely would be nice for these particular messages to be logged by the server so it'd be more obvious what is wrong. 

Thanks for your work on Ruff!

**Comment by dhruvmanila** [comment]
This issue should be fixed in the latest VS Code release (`2025.0.0`) when using Ruff >= `0.9.0`:

```
   0.017699542s  INFO main ruff_server::session::index: Registering workspace: /Users/dhruv/playground/ruff
   0.018302459s  INFO ruff:main ruff_server::server: Configuration file watcher successfully registered
   6.455773584s ERROR ThreadId(30) ruff_server::session::index::ruff_settings: Failed to parse /Users/dhruv/playground/ruff/pyproject.toml: TOML parse error at line 22, column 1
   |
22 | [tool.ruff.lint]
   | ^^^^^^^^^^^^^^^^
Unknown rule selector: ``

```

Resolved in https://github.com/astral-sh/ruff-vscode/pull/659.
